controllers/front/ProductController.php

```diff
@@ -890,7 +890,8 @@ protected function formatQuantityDiscounts($specific_prices, $price, $tax_rate,
                 // The price may be directly set
 
                 /** @var float $currentPriceDefaultCurrency current price with taxes in default currency */
-                $currentPriceDefaultCurrency = (!$row['reduction_tax'] ? $row['price'] : $row['price'] * (1 + $tax_rate / 100)) + (float) $ecotax_amount;
+                $currentPriceDefaultCurrency = $this->calculatePriceDefaultCurrency($row, $tax_rate, $ecotax_amount);
```

Comment:
@matthieu-rolland You should send values in parameters and not array. It's not the responsibility of the function to check is the value is in the array.

Reply:
done !